### PR TITLE
chore: Addressing TODO to load spec files directly for runner e2e tests

### PR DESCRIPTION
**Pull request by tbiethman**
- Part of UNIFY-1151

The runner e2e tests utilize a helper to open a project, launch the app, and execute a spec so that the test can assert the results of the spec run. Today, this helper navigates to a spec from the Specs page by searching for the spec name and clicking on it, which takes a bit of time.

However, with the completion of #20558, we are able to use `visitApp` to directly navigate to a spec without the app hanging. This removes some overhead for each spec we load and reduces test time run-app-integration by about ~30 seconds. Not a huge win, but it'll add up.

I also removed our usage of `__incorrectlyVisitAppWithIntercept` in the helper and replaced the interceptions with stubbing where appropriate.

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/cypress/e2e/runner/issues/issue-18042.cy.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { loadSpec } from '../support/spec-loader'
 describe('issue 18042', () => {
   it('Call count is shown even if cy.stub().log(false)', function () {
     loadSpec({
-      fileName: 'issue-18042.cy.js',
+      filePath: 'issues/issue-18042.cy.js',
```
Comment (tbiethman) [inline packages/app/cypress/e2e/runner/issues/issue-18042.cy.js:7]: The change from `fileName` -> `filePath` for the helper was made to ensure we could navigate to the appropriate url. Prior to this, selecting the fileName from the list accomplished this for us.

File: packages/app/cypress/e2e/runner/runner.ui.cy.ts [hunk 2]
```diff
@@ -157,19 +157,21 @@ describe('src/cypress/runner', () => {
...
+        filePath: 'runner/simple-cy-assert.runner.cy.js',
         passCount: 1,
         failCount: 0,
         hasPreferredIde: true,
       })
 
-      cy.intercept('mutation-SpecRunnerOpenMode_OpenFileInIDE', { data: { 'openFileInIDE': true } }).as('OpenIDE')
+      cy.withCtx((ctx, o) => {
+        ctx.actions.file.openFile = o.sinon.stub()
+      })
```
Comment (tbiethman) [inline packages/app/cypress/e2e/runner/runner.ui.cy.ts:168]: The helper is not using `__incorrectlyVisitAppWithIntercept` anymore, so replaced the intercept with appropriate stubbing.

File: packages/app/cypress/e2e/runner/support/verify-failures.ts [hunk 3]
```diff
@@ -35,6 +37,8 @@ const verifyFailure = (options) => {
     fileName,
     uncaught = false,
     uncaughtMessage,
+    ideLine,
+    ideColumn,
```
Comment (tbiethman) [inline packages/app/cypress/e2e/runner/support/verify-failures.ts:41]: Additionally verifying the line/column for the opened file where appropriate.

File: packages/app/cypress/e2e/runner/support/spec-loader.ts [hunk 4]
```diff
@@ -55,26 +57,13 @@ export function loadSpec (options: LoadSpecOptions): void {
...
 
-  cy.findByLabelText('Search Specs').type(fileName)
-  // wait for virtualized spec list to update, there is a chance
-  // of disconnection otherwise
-  cy.wait(500)
-  cy.contains('[data-cy=spec-item]', fileName).click()
-
-  cy.location().should((location) => {
-    expect(location.hash).to.contain(fileName)
-  })
```
Comment (flotwig) [inline packages/app/cypress/e2e/runner/support/spec-loader.ts:76]: ?

File: packages/app/cypress/e2e/runner/support/spec-loader.ts [hunk 5]
```diff
@@ -55,26 +57,13 @@ export function loadSpec (options: LoadSpecOptions): void {
     })
   }, { hasPreferredIde })
 
-  // TODO: investigate why directly visiting the spec will sometimes hang
-  // https://cypress-io.atlassian.net/browse/UNIFY-1154
```
Comment (flotwig) [inline packages/app/cypress/e2e/runner/support/spec-loader.ts:59]: were you able to validate that the flake is probably gone after the changes in this file?

**Review by flotwig — APPROVED** [review]
looks good, had a question about validation

File: packages/app/cypress/e2e/runner/support/spec-loader.ts (hunk 5 above)
Comment (tbiethman) [reply packages/app/cypress/e2e/runner/support/spec-loader.ts:59]: Yup, haven't seen a failure locally nor after numerous CI builds. This used to fail quite consistently before. There's more info in the PR description, but #20558 corrected the issue that was previously causing the hang.

**Review by lmiller1990 — APPROVED** [review]
Nice! Saving 30s on a 630 (5m 30s) run is a huge improvement.

**Pull request merged**